dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassPathEntry.java

```diff
@@ -96,27 +103,31 @@ public String toString() {
     }
   }
 
-  @VisibleForTesting
-  public static ClassPathEntry of(String coordinates, String filePath) {
-    Artifact artifact = new DefaultArtifact(coordinates);
-    return new ClassPathEntry(artifact.setFile(new File(filePath)));
-  }
-
   /**
    * Returns a list of class file names in {@link #jar} as in {@link JavaClass#getFileName()}. This
    * class file name is a path ("." as element separator) that locates a class file in a class path.
    * Usually the class name and class file name are the same. However a class file name may have a
    * framework-specific prefix. Example: {@code BOOT-INF.classes.com.google.Foo}.
    */
-  ImmutableSet<String> listClassFileNames() throws IOException {
-    URL jarUrl = jar.toUri().toURL();
+  // Could do this on construction but that makes construction slow and throw an IOException.
+  // Is this OK? Or maybe lazy load in getClassFileNames?
+  public void loadClassFileNames() throws IOException {
+    URL jarUrl = getJar().toUri().toURL();
     // Setting parent as null because we don't want other classes than this jar file
     URLClassLoader classLoaderFromJar = new URLClassLoader(new URL[] {jarUrl}, null);
 
     // Leveraging Google Guava reflection as BCEL doesn't have API to list classes in a jar file
     com.google.common.reflect.ClassPath classPath =
         com.google.common.reflect.ClassPath.from(classLoaderFromJar);
 
-    return classPath.getAllClasses().stream().map(ClassInfo::getName).collect(toImmutableSet());
+    classFileNames =
+        classPath.getAllClasses().stream().map(ClassInfo::getName).collect(toImmutableSet());
+  }
+  
+  /**
+   * @return the fully qualified names of the classes in this entry's jar file
+   */
+  public ImmutableSet<String> getClassNames() {
```

Comment:
How about calling `loadClassFileNames` and cache the result?

Reply:
I tried this. The problem is it introduces an IOException on this method, which both feels wrong and makes a lot of calling code very ugly. 

I'm not sure how to proceed. Suggestions appreciated.

Reply:
Maybe the right place to do this is in the constructor. It's called much less frequently than `getClassNames` so adding an IOException there would be less invasive.